Issue5 login view

develiasdaniel opened the pull request:
# Description
What did you implemented:
 - Design of log in view. 
 - atomization of components 
 - Implementation of component depending on the view
-  SingUp/logIn function depending on the view
 
 Examples:
 - What?: We added props in the parent component 
 - Why?: These changes complete the user login and account creation experience. See #7 .
 
 # Testing
- Download project 
- open terminal and
- run npm install
- run npx react-native start
- run npx react-native run-android or npx react-native run-ios 
 
 Example:
  - I've added coverage for testing all new methods. I used Faker for a few random user emails and names.
  
  ## Screenshots 
If applicable, include screenshots of the results or screenshots that help to see changes
![imagen](https://user-images.githubusercontent.com/69357302/98030360-84041900-1dd6-11eb-9037-f0444e37b33f.png)
![imagen](https://user-images.githubusercontent.com/69357302/98030654-f7a62600-1dd6-11eb-8d90-b0b0dd15a0b6.png)
![imagen](https://user-images.githubusercontent.com/69357302/98030475-a9912280-1dd6-11eb-8271-946b4eee31fa.png)
![imagen](https://user-images.githubusercontent.com/69357302/98030505-b9a90200-1dd6-11eb-97b3-4f389cab1a99.png)

tovl1 committed:
initial project

Commit:
adding iOS files

alcamk committed:
Merge pull request #5 from Brightcoders-Bootcamps/iss2-initialProject

initial project

Commit:
first approach to sign up view
co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

develiasdaniel committed:
add inputs and checks box of sign up form

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

NanciGarcia committed:
creation buttons Sign Up

Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>
Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>

NanciGarcia committed:
update style buttons

Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>
Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>

develiasdaniel committed:
add design button's of Sign Up view

Co-Authored-By: 1ng-HeberH <ing.heberh@gmail.com>
Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

Commit:
validating null form and adding or tag between buttons, also changing the touchable area to the full buttons
co-authored-by: NanciGarcia nancilizbethgarciamartines@gmail.com
co-authored-by: Oswaldotovar32 oswaldotovar32@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com

Commit:
adding the login screens

tovl1 committed:
Add disable functionality and log in tag
Co-Authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-Authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-Authored-by: JordiGN jordign.mx@gmail.com
Co-Authored-by: nancilizbethgarciamartinez@gmail.com

tovl1 committed:
Merge branch 'iss3-signupView' of https://github.com/Brightcoders-Bootcamps/kata-05-reservacion-vuelo-kata-05-team-02 into iss3-signupView

develiasdaniel committed:
add change color in Sign In Typing

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

NanciGarcia committed:
change color in the input password

Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>
Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>

tovl1 committed:
modify styles

Co-Authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-Authored-by: JordiGN jordign.mx@gmail.com
Co-Authored-by: nancilizbethgarciamartinez@gmail.com

Commit:
adding simulations for loader screen

co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-Authored-by: NanciGaria nancilizbethgarciamartinez@gmail.com
Co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

develiasdaniel committed:
add check image in progress circular bar

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

Commit:
adding modal backgournd opacity and separating components into individual files (except terms)

C-Co-Authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-Authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
Co-Authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

NanciGarcia committed:
Creations component terms

Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>
Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>

Commit:
adding an S at the end of a file name

Co-Authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
Co-Authored-by: oswaldotovar32 oswaldotovar32@gmail.com
Co-Authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com

tovl1 committed:
Merge branch 'iss3-signupView' into master

Merge iss3-signupView to master. Next merge to sandbox.

Commit:
adding package json to new brnach

co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

develiasdaniel committed:
add inputs en log in form

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

NanciGarcia committed:
Cretion of mainScreen

Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>
Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>

Commit:
Separating functionality for buttons, also moved the props to the mainScreen so both loginview and singupview can acces them

co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: develiasdanielsanchez eliasdanielsanchezlopez@gmail.com
co-authored-by: oswaldotovar32 oswaldotovar32@gmail.com

develiasdaniel committed:
add simulated sign up or login in

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

alcamk commented:
Good job team!

Ocosmee reviewed (dismissed):
Excellent job

Ing-Heber reviewed (dismissed):
Nice job, seems clean, notice the google icon its not visible at all when the login with google button changes to colour blue.

tovl1 committed:
add function to signup in firebase

Co-Authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-Authored-by: JordiGN jordign.mx@gmail.com
Co-Authored-by: nancilizbethgarciamartinez@gmail.com

NanciGarcia committed:
restructured valid email props

Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>
Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>

Commit:
adjusted the sign in function to be called just when the user email does not exist, and validating the errro messages from the sign in

co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com{

alcamk commented:
Look good for me, nice job!

alcamk reviewed (dismissed):
Look good, nice job!

NanciGarcia committed:
add function Sign Up with Count Google

Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>
Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>

NanciGarcia committed:
fix conflicts

Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>
Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>

Commit:
Adding configuration files for using Google/Firebase sign in

co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

develiasdaniel committed:
add configuration files for SignIn with google/firebase in android

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

Commit:
changin the bundle for GoogleService info-plist

develiasdaniel committed:
add sign in with google method

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

luisarmandodv commented on FlightReservation/components/termsComponents.js:
```diff
@@ -0,0 +1,66 @@
+      },
+    });
+    
+    export default Terms;
```
add newline at the end of the file

luisarmandodv commented on FlightReservation/components/termsComponents.js:
```diff
@@ -0,0 +1,66 @@
+  }
+
+  const styles = StyleSheet.create({
+
```
```suggestion
```

luisarmandodv commented on FlightReservation/components/termsComponents.js:
```diff
@@ -0,0 +1,66 @@
+
+
+function Terms(props) {
+    const { changeForm, objValues } = props;
+    return (
+      <View style={[styles.containerForm, { marginTop: 30 }]}>
```
indentation here has 4 spaces, but inside the JSX it has 2 spaces. Make a consistent spacing among the project is importante

luisarmandodv commented on FlightReservation/components/termsComponents.js:
```diff
@@ -0,0 +1,66 @@
+  TouchableOpacity,
+} from 'react-native';
+import CheckBox from '@react-native-community/checkbox';
+
```
Two spaces, not necessary here
```suggestion
```

luisarmandodv commented on FlightReservation/components/termsComponents.js:
```diff
@@ -0,0 +1,66 @@
+            value={objValues.subscribed}
+            onValueChange={(value) => changeForm('subscribed', value)}
+            style={styles.checkbox}
+            tintColors={{ true: '#3E59F7', false: 'black' }}
```
line 24 and line 35 contains the same styling, please do a Stylesheet object and use it on those lines.

luisarmandodv commented on FlightReservation/components/signupView.js:
```diff
@@ -0,0 +1,47 @@
+  containerForm: {
+    marginHorizontal: 15,
+  },
+
```
```suggestion
```

luisarmandodv commented on FlightReservation/components/signupView.js:
```diff
@@ -0,0 +1,47 @@
+
+
+function SignupForm(props) {
+  const {setIsLoginFormActive,isLoginFormActive,showObj, showModal, addFill, formObjectState, modalVisible, singedText, isIconCheck} = props;
```
this line is horizontally really long, split it like this:
```suggestion
  const {
    setIsLoginFormActive,
    isLoginFormActive,
    showObj,
    showModal,
    addFill,
    formObjectState,
    modalVisible,
    singedText,
    isIconCheck,
    } = props;
```

luisarmandodv commented on FlightReservation/components/signupView.js:
```diff
@@ -0,0 +1,47 @@
+import SignUpButton from './signUpButtons';
+import Form from './formComponent';
+import Terms from './termsComponents';
+
+
```
```suggestion
```

luisarmandodv commented on FlightReservation/components/signUpButtons.js:
```diff
@@ -0,0 +1,126 @@
+    },
+});
+
+export default SignUpButton;
```
add break-line at the end of the file

luisarmandodv commented on FlightReservation/components/signUpButtons.js:
```diff
@@ -0,0 +1,126 @@
+    TouchableOpacity,
+} from 'react-native';
+
+function SignUpButton(props) {
```
All this file contains 4 spaces of indentation. But on other components, you are using 2 spaces. You should decide which on to use consistently among all the project.

luisarmandodv commented on FlightReservation/components/modalComponent.js:
```diff
@@ -0,0 +1,93 @@
+  }
+  });
+
+  export default ModalCustom;
```
add break-line at the end of this file

luisarmandodv commented on FlightReservation/components/modalComponent.js:
```diff
@@ -0,0 +1,93 @@
+  }
+  
+  const styles = StyleSheet.create({
+      //Estilos de modal
```
```suggestion
```

luisarmandodv commented on FlightReservation/components/modalComponent.js:
```diff
@@ -0,0 +1,93 @@
+        >
+          <View style={[styles.centeredView, {backgroundColor:'rgba(0, 0, 0, 0.4)' }]}>
+            <View style={[styles.modalView, {backgroundColor: "black"}]}>
+              {isIconCheck
+                ? <AnimatedCircularProgress
+                  size={80}
+                  width={8}
+                  fill={100}
+                  tintColor="#5362D8"
+                  backgroundColor="#3d5875"
+                  duration={2500}
+                />
+                : <Image
+                  style={styles.iconCheck}
+                  source={require('../img/check.png')}
+                />
+              }
```
instead of having this ternary condition, it would be better to call a function with the return of this JSX

luisarmandodv commented on FlightReservation/components/mainScreen.js:
```diff
@@ -0,0 +1,93 @@
+  }
+
+  return (
+
+    <>
+      {isLoginFormActive
+        ? <LoginView isLoginFormActive={isLoginFormActive} setIsLoginFormActive={setIsLoginFormActive} showObj={showObj}
+          showModal={showModal} addFill={addFill} formObjectState={formObjectState} modalVisible={modalVisible} singedText={singedText} isIconCheck={isIconCheck} />
+
+        : <SignupForm setIsLoginFormActive={setIsLoginFormActive} isLoginFormActive={isLoginFormActive} setIsLoginFormActive={setIsLoginFormActive} showObj={showObj}
+          showModal={showModal} addFill={addFill} formObjectState={formObjectState} modalVisible={modalVisible} singedText={singedText} isIconCheck={isIconCheck} />
+      }
+
+    </>
```
- remove extra break-lines
- those lines are really long, please use `Prettier ` on those lines for better formating

luisarmandodv commented on FlightReservation/components/mainScreen.js:
```diff
@@ -0,0 +1,93 @@
+  }
+
+  const SignUpOrLoginAction = () =>{
+    if(isLoginFormActive){
+      //accion para login
+      console.log("logeado")
+    }else {
+      console.log("registrado")
+    }
```
this function is useless

JordiGN replied:
It is just a test for components communication, next issue use this function

luisarmandodv commented on FlightReservation/components/mainScreen.js:
```diff
@@ -0,0 +1,93 @@
+  },[isLoginFormActive])
+
+  const showObj = () => {
+    console.log(formObjectState);
```
```suggestion
```

luisarmandodv commented on FlightReservation/components/mainScreen.js:
```diff
@@ -0,0 +1,93 @@
+
+import SignupForm from './signupView';
+import LoginView from './loginView';
+
```
```suggestion
```

luisarmandodv commented on FlightReservation/components/loginView.js:
```diff
@@ -0,0 +1,39 @@
+import ModalCustom from './modalComponent';
+
+function LoginView(props) {
+  const {setIsLoginFormActive, isLoginFormActive, showObj, showModal, addFill, formObjectState, modalVisible, singedText, isIconCheck} = props;
```
make this line shorter

luisarmandodv commented on FlightReservation/components/formComponent.js:
```diff
@@ -0,0 +1,177 @@
+    },
+});
+
+export default Form;
```
add a break-line here

luisarmandodv commented on FlightReservation/components/formComponent.js:
```diff
@@ -0,0 +1,177 @@
+    Pressable,
+} from 'react-native';
+
+function Form(props) {
```
check identation here, 4 lines or 2 lines?

luisarmandodv reviewed (changes requested):
- I strongly recommend the use of `Prettier` VS Code module, in order to handle the code formating easily.
- All components must contain `PropTypes` defined on them, please review: https://es.reactjs.org/docs/typechecking-with-proptypes.html

Commit:
adding google sign up on iOS, and also registering the data to the users table

co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com

Commit:
Deleting user info when is wrong, adjusting timing for modal and laoders, validating repeated info

co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

Commit:
Solving Code review requests

co-authored-by: NanciGarcia naniclizbethgarciamartinez@gmail.com
Co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

JordiGN reviewed (dismissed):
Solving changes

luisarmandodv commented on FlightReservation/components/mainScreen.js:
```diff
@@ -0,0 +1,96 @@
+import React, {useState} from 'react';
+import {View, Text} from 'react-native';
+
```
```suggestion
```

JordiGN replied:
done

JordiGN replied:
We already corrected all the suggestions you have done to us. Except for the one about "proptypes", as we already continue with another issue and a pull request, and that one has more changes, and obviously the same issues, we are going to add the "proptypes" in that issue, so we would try to avoid merge conflicts.

luisarmandodv commented on FlightReservation/components/signUpButtons.js:
```diff
@@ -0,0 +1,134 @@
+            ? 'Do not have an account?'
+            : 'Already have an account'}{' '}
+        </Text>
+
```
```suggestion
```

JordiGN replied:
Done

luisarmandodv commented on FlightReservation/App.js:
```diff
@@ -0,0 +1,29 @@
+  StatusBar,
+} from 'react-native';
+
+import MainScreen from './components/mainScreen';
```
it would be better to have a folder with the `screens` and a folder with the `components`.

JordiGN replied:
thank you for your suggestions, we already applied them, check it on the next pull request.

luisarmandodv reviewed (changes requested):
I don't see any PropType defined anywhere

Commit:
solving changes requested on Pull request #5

Co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
Co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

Commit:
merging issue5 and issue4 into a temporal branch in order to solve merge conlicts

Co-authorerd-by: NanciGarcia nancilizbethgarciamartienz@gmail.com
Co-authorerd-by:OswaldoTovar32 oswaldotovar32@gmail.com
Co-authorerd-by: develiasdaniel eliasdanielsanchezlopez@gmail.com

luisarmandodv commented on FlightReservation/components/formComponent.js:
```diff
@@ -0,0 +1,188 @@
+      {!isLoginFormActive && (
+        <View>
+          <Text style={styles.inputHeader}>First Name</Text>
+          <TextInput
+            style={[
+              styles.inputStyle,
+              {borderColor: borderColor.borderColorName},
+            ]}
+            onChange={(e) =>
+              checkInputsIsNull(e, 'name', 'borderColorName')
+            }></TextInput>
```
```suggestion
          <TextInput
            style={[
              styles.inputStyle,
              {borderColor: borderColor.borderColorName},
            ]}
            onChange={(e) =>
              checkInputsIsNull(e, 'name', 'borderColorName')
            } />
```

luisarmandodv commented on FlightReservation/components/formComponent.js:
```diff
@@ -0,0 +1,188 @@
+          </Text>
+        )}
+      </View>
+      <TextInput
+        style={[styles.inputStyle, {borderColor: borderColor.borderColorEmail}]}
+        onChange={(e) =>
+          checkInputsIsNull(e, 'email', 'borderColorEmail')
+        }></TextInput>
```
```suggestion
      <TextInput
        style={[styles.inputStyle, {borderColor: borderColor.borderColorEmail}]}
        onChange={(e) =>
          checkInputsIsNull(e, 'email', 'borderColorEmail')
        }/>
```

luisarmandodv commented on FlightReservation/components/formComponent.js:
```diff
@@ -0,0 +1,188 @@
+            checkInputsIsNull(e, 'password', 'borderColorPassword')
+          }
+        />
+
```
```suggestion
```

luisarmandodv commented on FlightReservation/components/modalComponent.js:
```diff
@@ -0,0 +1,95 @@
+        duration={2500}
+      />
+    );
+  } else {
+    return (
+      <Image style={styles.iconCheck} source={require('../img/check.png')} />
+    );
+  }
```
```suggestion
  }
  return (
    <Image style={styles.iconCheck} source={require('../img/check.png')} />
  );
```

luisarmandodv reviewed (changes requested):
small comments and they seem to be the last ones

develiasdaniel committed:
refactoring changes suggested in the pr #4

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

develiasdaniel committed:
add changes suggested in pr #4

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

develiasdaniel committed:
add prop types

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

Commit:
deleting an unnecesary else statement

Co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
Co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

Ocosmee reviewed (approved):
Good job!

alcamk reviewed (approved):
I think is a nice view and it's functional, so it's good for me.

Ing-Heber reviewed (approved):
Ojitoooo, looks nice, clean and understandable, nice component implementation and fb config. Well done team!
